pkgs/ios/Metriport-IOS/Sources/Metriport-IOS/HealthStore.swift

```diff
@@ -0,0 +1,71 @@
+import Foundation
+import HealthKit
+import Combine
+import CoreData
+import WebKit
+
+public class MetriportHealthStoreManager {
+    public let healthStore = HKHealthStore()
+    public let metriportClient: MetriportClient
+    private let healthKitTypes = HealthKitTypes()
+    private var metriportUserId = ""
+
+    public init() {
+        self.metriportClient = MetriportClient(healthStore: healthStore)
+        
+        // If we've already authorized then start checking background updates on app load
+        if UserDefaults.standard.object(forKey: "HealthKitAuth") != nil {
```

Comment:
What happens if the user manually revokes the permissions in settings?

Should we always be checking if permissions are actually still granted just in case?

Reply:
So the HealthKitAuth key is just to verify that the user has been shown the screen to access the different data types.

The reason we have it is so we dont make requests to healthkit before being that.

What the user can do is remove access to certain data types (or all of them). What will happen then is we just wont receive that data but no errors will be thrown.